src/hotspot/share/gc/shared/space.cpp

```diff
@@ -147,13 +147,11 @@ bool Space::obj_is_alive(const HeapWord* p) const {
 
 void ContiguousSpace::object_iterate(ObjectClosure* blk) {
   if (is_empty()) return;
-  object_iterate_from(bottom(), blk);
-}
-
-void ContiguousSpace::object_iterate_from(HeapWord* mark, ObjectClosure* blk) {
-  while (mark < top()) {
-    blk->do_object(cast_to_oop(mark));
-    mark += cast_to_oop(mark)->size();
+  HeapWord* addr = bottom();
+  while (addr < top()) {
+    oop obj = cast_to_oop(addr);
+    blk->do_object(obj);
+    addr += obj->size();
   }
 }
```

Comment:
Given that this function now grows beyond a small two-liner implementation, I would prefer if the `is_empty`check was separated from the rest of the code. Something like this:
```
void ContiguousSpace::object_iterate(ObjectClosure* blk) {
  if (is_empty()) {
    return;
  }
  
  HeapWord* addr = bottom();
  while (addr < top()) {
    oop obj = cast_to_oop(addr);
    blk->do_object(obj);
    addr += obj->size();
  }
}
```

With that said, it seems like the `is_empty()` boils down to a comparison between `bottom()` and `top()`. If that is the case, then we could just remove the `is_empty()` check and let that be handled by the while loop.